src/main/java/htsjdk/samtools/util/BufferedLineReader.java

```diff
@@ -41,6 +41,25 @@
  */
 public class BufferedLineReader extends LineNumberReader implements LineReader {
 
+    private static class StringBackedInputStream extends InputStream {
+        private int idx = 0;
+        private final String str;
+        private final int len;
+
+        StringBackedInputStream(String str) {
+            this.str = str;
+            this.len = str.length();
+        }
+
+        @Override
+        public int read() throws IOException {
+            if(idx >= len) {
+                return -1;
+            }
+            return (byte) str.charAt(idx++);
```

Comment:
I might be confused about the results of charAt, but won't this mangle non-ascii characters?  We already do mangle non-ascii in a lot of places, but I would prefer we don't add more instances of it.  SAM headers can contain UTF-8 in a number of places.

Reply:
@nh13 Any comment on this?

Reply:
It should work with UTF-8. 

```scala
scala> val string =  "ﾖ"
string: String = ﾖ

scala> string.charAt(0)
res0: Char = ﾖ

scala> string.length
res1: Int = 1
```

Reply:
I meant that casting from char -> byte would cause problems with utf-8 since we'd be losing the second byte of information.

Reply:
I was unclear was unclear in my writing.

Reply:
True.  Any suggestions on how to make it work.  PRs welcome as I'm about to go on vacation.

```scala
scala> 'ﾖ'.isValidByte
res0: Boolean = false

scala> 'ﾖ'.isValidInt
res1: Boolean = true
```

Reply:
Changing `byte` to `int`